llama_index/storage/docstore/simple_docstore.py

```diff
@@ -47,7 +47,10 @@ def from_persist_dir(
 
         """
 
-        persist_path = os.path.join(persist_dir, DEFAULT_PERSIST_FNAME)
+        if fs is not None:
+            persist_path = persist_dir + "/" + DEFAULT_PERSIST_FNAME
```

Comment:
This feels kind of unreliable. I wish fsspec provided something to join paths automatically 🤔 I'm stuck on the best way to handle this. I will need to read more about fsspec tbh

Reply:
Hey @logan-markewich , any update on this? Without this fix the S3 integration (and possibly other cloud storages) aren't usable on Windows.

Reply:
Sorry for the delay on this @yoeldk. I had looked into fsspec but they don't provide a solution for this

I think for now, can you create a utility function inside `llama_index/utils.py` and use that, instead of concatenating a "/" character? Just want to automatically handle cases where you join with/without trailing/leading "/" characters (i.e. joining `my/path` to `dir_name` or `/my/path/` and `./dir_name` should produce similar results without doubling up "/" characters

Also you'll have to merge `main` into your branch, there are some conflicts.

Reply:
Hey @logan-markewich . I updated the branch with a simple fix. Apparently os.path.join(a,b) on Windows will add a backslash before b if a doesn't end with a slash. So my fix is to add a trailing slash to a (if it doesn't have one) and then use os.path.join.
I pushed my changes to the repository so the PR should now contain it.
Hopefully now it's OK :)

Reply:
Looks good to me, thanks for your patience on this haha!